extensions/oidc-devservices-keycloak/deployment/src/main/java/io/quarkus/devservices/keycloak/deployment/KeycloakDevServicesUtils.java

```diff
@@ -0,0 +1,74 @@
+package io.quarkus.devservices.keycloak.deployment;
+
+import io.quarkus.oidc.common.runtime.OidcCommonUtils;
+import io.vertx.core.MultiMap;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.json.JsonObject;
+import io.vertx.mutiny.core.buffer.Buffer;
+import io.vertx.mutiny.ext.web.client.HttpRequest;
+import io.vertx.mutiny.ext.web.client.HttpResponse;
+import io.vertx.mutiny.ext.web.client.WebClient;
+
+public final class KeycloakDevServicesUtils {
+    private KeycloakDevServicesUtils() {
+
+    }
+
+    public static WebClient createWebClient() {
+        Vertx vertx = Vertx.vertx();
+        return WebClient.create(new io.vertx.mutiny.core.Vertx(vertx));
+    }
+
+    public static String getPasswordAccessToken(WebClient client,
+            String keycloakUrl,
+            String clientId,
+            String clientSecret,
+            String userName,
+            String userPassword) throws Exception {
+        HttpRequest<Buffer> request = client.postAbs(keycloakUrl);
+        request.putHeader(HttpHeaders.CONTENT_TYPE.toString(), HttpHeaders.APPLICATION_X_WWW_FORM_URLENCODED.toString());
+
+        io.vertx.mutiny.core.MultiMap props = new io.vertx.mutiny.core.MultiMap(MultiMap.caseInsensitiveMultiMap());
+        props.add("client_id", clientId);
+        if (clientSecret != null) {
+            props.add("client_secret", clientSecret);
+        }
+
+        props.add("username", userName);
+        props.add("password", userPassword);
+        props.add("grant_type", "password");
+
+        return request.sendBuffer(OidcCommonUtils.encodeForm(props)).onItem()
+                .transform(resp -> getAccessTokenFromJson(resp)).await().indefinitely();
```

Comment:
This should likely have some form of timeout. No sure if you can set a request timeout in the vert.x client itself?

Reply:
@stuartwdouglas I could not find it in `WebClientOptions` but this timeout is now configurable and I've updated the code to refer to it